### ISSUE TITLE
strategy: free -- Output not as expected

### ISSUE THREAD

**Issue by x12Mike** [ansible/ansible#14027]
When running a playbook with "strategy: free" set, the output of the playbook doesn't line up.

Example Playbook:

```
    - hosts: "srv01:srv02:srv03"
      strategy: free
      sudo: yes

      tasks:
      - name: Task 1 - Say Hello
        shell: echo "HELLO WORLD!"

      - name: Task 2 - Echo Date
        shell: date

      - name: Task 3 - Say Bye
        shell: echo "GOODBYE!"
```

Output on command-line:

```
PLAY ***************************************************************************

TASK [Task 1 - Say Hello] ******************************************************

TASK [Task 1 - Say Hello] ******************************************************

TASK [Task 1 - Say Hello] ******************************************************
changed: [srv02] => {"changed": true, "cmd": "echo \"HELLO WORLD!\"", "delta": "0:00:00.004532", "end": "2016-01-20 10:33:20.762749", "rc": 0, "start": "2016-01-20 10:33:20.758217", "stderr": "", "stdout": "HELLO WORLD!", "stdout_lines": ["HELLO WORLD!"], "warnings": []}

TASK [Task 2 - Echo Date] ******************************************************
changed: [srv01] => {"changed": true, "cmd": "echo \"HELLO WORLD!\"", "delta": "0:00:00.004590", "end": "2016-01-20 13:33:20.802488", "rc": 0, "start": "2016-01-20 13:33:20.797898", "stderr": "", "stdout": "HELLO WORLD!", "stdout_lines": ["HELLO WORLD!"], "warnings": []}
changed: [srv03] => {"changed": true, "cmd": "echo \"HELLO WORLD!\"", "delta": "0:00:00.005041", "end": "2016-01-20 13:33:20.818528", "rc": 0, "start": "2016-01-20 13:33:20.813487", "stderr": "", "stdout": "HELLO WORLD!", "stdout_lines": ["HELLO WORLD!"], "warnings": []}

TASK [Task 2 - Echo Date] ******************************************************

TASK [Task 2 - Echo Date] ******************************************************
changed: [srv02] => {"changed": true, "cmd": "date", "delta": "0:00:00.006498", "end": "2016-01-20 10:33:22.189501", "rc": 0, "start": "2016-01-20 10:33:22.183003", "stderr": "", "stdout": "Wed Jan 20 10:33:22 PST 2016", "stdout_lines": ["Wed Jan 20 10:33:22 PST 2016"], "warnings": []}
changed: [srv03] => {"changed": true, "cmd": "date", "delta": "0:00:00.005979", "end": "2016-01-20 13:33:22.213036", "rc": 0, "start": "2016-01-20 13:33:22.207057", "stderr": "", "stdout": "Wed Jan 20 13:33:22 EST 2016", "stdout_lines": ["Wed Jan 20 13:33:22 EST 2016"], "warnings": []}

TASK [Task 3 - Say Bye] ********************************************************

TASK [Task 3 - Say Bye] ********************************************************
changed: [srv01] => {"changed": true, "cmd": "date", "delta": "0:00:00.006046", "end": "2016-01-20 13:33:22.224040", "rc": 0, "start": "2016-01-20 13:33:22.217994", "stderr": "", "stdout": "Wed Jan 20 13:33:22 EST 2016", "stdout_lines": ["Wed Jan 20 13:33:22 EST 2016"], "warnings": []}

TASK [Task 3 - Say Bye] ********************************************************
changed: [srv02] => {"changed": true, "cmd": "echo \"GOODBYE!\"", "delta": "0:00:00.004980", "end": "2016-01-20 10:33:23.536479", "rc": 0, "start": "2016-01-20 10:33:23.531499", "stderr": "", "stdout": "GOODBYE!", "stdout_lines": ["GOODBYE!"], "warnings": []}
changed: [srv03] => {"changed": true, "cmd": "echo \"GOODBYE!\"", "delta": "0:00:00.004928", "end": "2016-01-20 13:33:23.543892", "rc": 0, "start": "2016-01-20 13:33:23.538964", "stderr": "", "stdout": "GOODBYE!", "stdout_lines": ["GOODBYE!"], "warnings": []}
changed: [srv01] => {"changed": true, "cmd": "echo \"GOODBYE!\"", "delta": "0:00:00.004510", "end": "2016-01-20 13:33:23.675290", "rc": 0, "start": "2016-01-20 13:33:23.670780", "stderr": "", "stdout": "GOODBYE!", "stdout_lines": ["GOODBYE!"], "warnings": []}

PLAY RECAP *********************************************************************
srv01                    : ok=3    changed=3    unreachable=0    failed=0   
srv02                    : ok=3    changed=3    unreachable=0    failed=0   
srv03                    : ok=3    changed=3    unreachable=0    failed=0
```

If you notice, the output of "srv01's Task 2", shows up under the "Task 3 - Say Bye" header.  My thoughts are that the output should show like this:

```
PLAY ***************************************************************************

TASK [Task 1 - Say Hello] ******************************************************
changed: [srv01] => {"changed": true, "cmd": "echo \"HELLO WORLD!\"", "delta": "0:00:00.004590", "end": "2016-01-20 13:33:20.802488", "rc": 0, "start": "2016-01-20 13:33:20.797898", "stderr": "", "stdout": "HELLO WORLD!", "stdout_lines": ["HELLO WORLD!"], "warnings": []}

TASK [Task 1 - Say Hello] ******************************************************
changed: [srv02] => {"changed": true, "cmd": "echo \"HELLO WORLD!\"", "delta": "0:00:00.004532", "end": "2016-01-20 10:33:20.762749", "rc": 0, "start": "2016-01-20 10:33:20.758217", "stderr": "", "stdout": "HELLO WORLD!", "stdout_lines": ["HELLO WORLD!"], "warnings": []}

TASK [Task 1 - Say Hello] ******************************************************
changed: [srv03] => {"changed": true, "cmd": "echo \"HELLO WORLD!\"", "delta": "0:00:00.005041", "end": "2016-01-20 13:33:20.818528", "rc": 0, "start": "2016-01-20 13:33:20.813487", "stderr": "", "stdout": "HELLO WORLD!", "stdout_lines": ["HELLO WORLD!"], "warnings": []}

TASK [Task 2 - Echo Date] ******************************************************
changed: [srv01] => {"changed": true, "cmd": "date", "delta": "0:00:00.006046", "end": "2016-01-20 13:33:22.224040", "rc": 0, "start": "2016-01-20 13:33:22.217994", "stderr": "", "stdout": "Wed Jan 20 13:33:22 EST 2016", "stdout_lines": ["Wed Jan 20 13:33:22 EST 2016"], "warnings": []}

TASK [Task 2 - Echo Date] ******************************************************
changed: [srv02] => {"changed": true, "cmd": "date", "delta": "0:00:00.006498", "end": "2016-01-20 10:33:22.189501", "rc": 0, "start": "2016-01-20 10:33:22.183003", "stderr": "", "stdout": "Wed Jan 20 10:33:22 PST 2016", "stdout_lines": ["Wed Jan 20 10:33:22 PST 2016"], "warnings": []}

TASK [Task 2 - Echo Date] ******************************************************
changed: [srv03] => {"changed": true, "cmd": "date", "delta": "0:00:00.005979", "end": "2016-01-20 13:33:22.213036", "rc": 0, "start": "2016-01-20 13:33:22.207057", "stderr": "", "stdout": "Wed Jan 20 13:33:22 EST 2016", "stdout_lines": ["Wed Jan 20 13:33:22 EST 2016"], "warnings": []}

TASK [Task 3 - Say Bye] ********************************************************
changed: [srv02] => {"changed": true, "cmd": "echo \"GOODBYE!\"", "delta": "0:00:00.004980", "end": "2016-01-20 10:33:23.536479", "rc": 0, "start": "2016-01-20 10:33:23.531499", "stderr": "", "stdout": "GOODBYE!", "stdout_lines": ["GOODBYE!"], "warnings": []}

TASK [Task 3 - Say Bye] ********************************************************
changed: [srv03] => {"changed": true, "cmd": "echo \"GOODBYE!\"", "delta": "0:00:00.004928", "end": "2016-01-20 13:33:23.543892", "rc": 0, "start": "2016-01-20 13:33:23.538964", "stderr": "", "stdout": "GOODBYE!", "stdout_lines": ["GOODBYE!"], "warnings": []}

TASK [Task 3 - Say Bye] ********************************************************
changed: [srv01] => {"changed": true, "cmd": "echo \"GOODBYE!\"", "delta": "0:00:00.004510", "end": "2016-01-20 13:33:23.675290", "rc": 0, "start": "2016-01-20 13:33:23.670780", "stderr": "", "stdout": "GOODBYE!", "stdout_lines": ["GOODBYE!"], "warnings": []}

PLAY RECAP *********************************************************************
srv01                    : ok=3    changed=3    unreachable=0    failed=0   
srv02                    : ok=3    changed=3    unreachable=0    failed=0   
srv03                    : ok=3    changed=3    unreachable=0    failed=0
```

**Comment by orderedchaosdev** [comment]
I also have issues where strategy free breaks my terminal and runs random commands

**Comment by jimi-c** [comment]
@x12Mike  this is something of a known issue, I had originally intended to write a custom callback to handle the free strategy so as to better handle the interleaved nature of the messages, so we'll work on fixing that in the near future.

**Comment by jimi-c** [comment]
@orderedchaosdev please open a new issue for that, as it's not related to the above issue.

**Comment by orderedchaosdev** [comment]
Ok will do

**Comment by x12Mike** [comment]
@jimi-c - Thank you!  I had noticed it today as I was testing the strategy feature.  It threw me for a loop because I couldn't determine which server was doing what.  I checked on the #ansible IRC channel to see if this was known, they directed me here to search if it was.  I hadn't seen anything with my searching, so I opted to file the issue.

**Comment by jimi-c** [comment]
@x12Mike no worries, we'll use this as the issue to track fixing it. Thanks!

**Comment by wysekm** [comment]
Is there any work being done on this? At the moment task and server names are displayed completely random. I don't believe that it's not bothering a lot of people. Can anything be done to move it forward?

**Comment by jimi-c** [comment]
@wysekm this is something we keep putting on the back burner. I'll make it a priority for 2.2.0.

**Comment by wysekm** [comment]
@jimi-c Thanks. It would be really great if it gets released with 2.2.0!

**Comment by bcoca** [comment]
closing via #17355
